src/Plugin/QuickForm/SaatenUnionSpraying.php

```diff
@@ -0,0 +1,427 @@
+        '@message' => $e->getMessage(),
+      ]);
+      return [];
+    }
+  }
+
+  /**
+   * Builds the form.
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['log_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Log Name'),
+      '#required' => TRUE,
+      '#description' => $this->t('Log name/description of event.'),
+    ];
+
+    $form['date_start'] = [
+      '#type' => 'datetime',
+      '#title' => $this->t('Operation start time and date'),
+      '#required' => TRUE,
+      '#description' => $this->t('The start date and time of the operation.'),
+    ];
+
+    $form['date_end'] = [
+      '#type' => 'datetime',
+      '#title' => $this->t('Operation finish time and date'),
+      '#required' => TRUE,
+      '#description' => $this->t('The finish date and time of the operation.'),
+    ];
+
+    $form['status'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Job Status'),
+      '#options' => [
+        'done' => $this->t('Done'),
+        'pending' => $this->t('Pending'),
+      ],
+      '#default_value' => 'done',
+      '#required' => TRUE,
+      '#description' => $this->t('The current status of the job.'),
+    ];
+
+    $form['flag'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Flag'),
+      '#options' => [
+        'monitor' => $this->t('Monitor'),
+        'priority' => $this->t('Priority'),
+        'needs_review' => $this->t('Needs Review'),
+      ],
+      '#multiple' => TRUE,
+      '#description' => $this->t('Flag this job if it is a priority, requires monitoring or review.'),
+    ];
+
+    // Users to assign.
+    $users = $this->getUserOptions($this->entityTypeManager);
+    $form['assigned_to'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Assigned to'),
+      '#options' => $users,
+      '#multiple' => TRUE,
+      '#required' => TRUE,
+      '#description' => $this->t('The operator(s) who carried out the task.'),
+    ];
+
+    $materials = $this->getMaterials();
+    $form['product'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Product'),
+      '#options' => $materials,
+      '#required' => TRUE,
+      '#description' => $this->t('The product used.'),
+    ];
+    
+    $form['product_rate'] = [
+      '#type' => 'entity_autocomplete',
+      '#title' => $this->t('Product Rate'),
+      '#description' => $this->t('The rate the product is applied per unit area.'),
+      '#target_type' => 'material_type',
+      '#required' => TRUE,
+    ];
+
+    $form['total_product_quantity'] = [
+      '#type' => 'entity_autocomplete',
+      '#title' => $this->t('Total Product Quantity'),
+      '#description' => $this->t('The total amount of product required to cover the field area(s).'),
+      '#target_type' => 'material',
+      '#selection_settings' => [
+        'target_bundles' => ['assets' => 'material'],
+      ],
+      '#required' => TRUE,
+    ];
```

Comment:
These will be quantities. Neither of these  should be `entity_autocomplete` and will not need `#target_type` or `#selection_settings`.

Reply:
I have made these updates and added the getQuantities function